runners/core-java/src/main/java/org/apache/beam/runners/core/DoFnRunners.java

```diff
@@ -144,5 +159,70 @@
         stepContext,
         aggregatorFactory,
         windowingStrategy);
+
+    if (!(doFn instanceof ReduceFnExecutor)) {
+      return doFnRunner;
+    } else {
+      // When a DoFn is a ReduceFnExecutor, we know it has to have an aggregator for dropped
+      // elements and we also learn that for some K and V,
+      //   InputT = KeyedWorkItem<K, V>
+      //   OutputT = KV<K, V>
+
+      Aggregator<Long, Long> droppedDueToLatenessAggregator =
+          ((ReduceFnExecutor<?, ?, ?, ?>) doFn).getDroppedDueToLatenessAggregator();
+
+      @SuppressWarnings({"unchecked", "cast", "rawtypes"})
+      DoFnRunner<InputT, OutputT> runner = (DoFnRunner<InputT, OutputT>) lateDataDroppingRunner(
+          (DoFnRunner) doFnRunner,
+          stepContext,
+          (WindowingStrategy) windowingStrategy,
+          droppedDueToLatenessAggregator);
+
+      return runner;
+    }
+  }
+
+  /**
+   * @deprecated for migration purposes only
```

Comment:
Can you clarify if this is deprecated because it is going away or deprecated as a sign its not ready yet? Maybe just say `@deprecated You should use ... instead.`